tests/frontend/specs/timeslider_follow.js

```diff
@@ -1,55 +1,69 @@
   //create a new pad before each test run
   beforeEach(function(cb){
     helper.newPad(cb);
-    this.timeout(6000);
   });
 
-  it("follow content as it's added to timeslider", function(done) { // passes
-    var inner$ = helper.padInner$;
-    var chrome$ = helper.padChrome$;
-
-    // make some changes to produce 100 revisions
-    var timePerRev = 900
-      , revs = 10;
-    this.timeout(revs*timePerRev+10000);
-    for(var i=0; i < revs; i++) {
-      setTimeout(function() {
-        // enter 'a' in the first text element
-        inner$("div").last().sendkeys('a\n');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-      }, timePerRev*i);
+  it("content as it's added to timeslider", async function() {
+    // send 3 revisions
+    let revs = 6;
+    let message = 'a\n\n\n\n\n\n\n\n\n\n';
+    let newLines = message.split('\n').length
+    for (let i=0;i<revs;i++){
+      await helper.edit(message, newLines*i + 1);
     }
 
-    setTimeout(function() {
-      // go to timeslider
-      $('#iframe-container iframe').attr('src', $('#iframe-container iframe').attr('src')+'/timeslider#0');
+    await helper.gotoTimeslider(0);
+    await helper.waitForPromise(function(){return helper.contentWindow().location.hash === '#0'})
 
-      setTimeout(function() {
-        var timeslider$ = $('#iframe-container iframe')[0].contentWindow.$;
-        var $sliderBar = timeslider$('#ui-slider-bar');
+    let originalTop = helper.contentWindow().$('#innerdocbody').offset();
 
-        var latestContents = timeslider$('#innerdocbody').text();
+    // set to follow contents as it arrives
+    helper.contentWindow().$('#options-followContents').prop("checked", true);
+    helper.contentWindow().$('#playpause_button_icon').click();
 
-        // set to follow contents as it arrives
-        timeslider$('#options-followContents').prop("checked", true);
+    let newTop;
+    return helper.waitForPromise(function(){
+      newTop = helper.contentWindow().$('#innerdocbody').offset();
+      return newTop.top < originalTop.top;
+    })
+  });
 
-        var originalTop = timeslider$('#innerdocbody').offset();
-        timeslider$('#playpause_button_icon').click();
+  it("only to lines that exist in the current pad view #4389", async function(){
+    let rev0text = helper.textLines()[0];
+    let rev1text = "\xa0"; // timeslider will add &nbsp;
+    let rev2text = 'Test line'
 
-        setTimeout(function() {
-          //make sure the text has changed
-          var newTop = timeslider$('#innerdocbody').offset();
-          expect( originalTop ).not.to.eql( newTop );
-          done();
-        }, 1000);
+    helper.padInner$('body').empty();
+    await helper.waitForPromise(function(){
+      return helper.commits.length === 1; 
+    })
+    await helper.edit(rev2text)
```

Comment:
It's unclear to me what this does. Does it:

  * add the text to the beginning of the existing line 1?
  * add the text to the end of the existing line 1?
  * replace the existing line 1?
  * insert a new line before the existing line 1?

Reply:
OK. Yes this is not clear. It will put the string into the pad, depending on the caret position and depending on the `lineNum` parameter (defaults to the first line).
It is enough for the tests here but I need to investigate how the other tests use sendkeys to find out what should be supported when "editing" a pad.